src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,160 @@
+  isLarge?: boolean
+  forceAction?: boolean
+  modalRoot?: string
+}
+
+export type ModalProps = ModalComponentProps &
+  Pick<ModalHook, 'isOpen' | 'closeModal'> &
+  JSX.IntrinsicElements['div']
+
+// TODO:
+// decide how to handle modal toggle button (anchor, button)
+//  - aria-controls=modal ID
+
+export const Modal = ({
+  id,
+  children,
+  isOpen,
+  closeModal,
+  isLarge = false,
+  forceAction = false,
+  modalRoot = '.usa-modal-wrapper',
+  ...divProps
+}: ModalProps): React.ReactElement => {
+  const [mounted, setMounted] = useState(false)
+  const initialPaddingRef = useRef<string>()
+  const tempPaddingRef = useRef<string>()
+  const modalEl = useRef<HTMLDivElement>(null)
+
+  const modalRootSelector = modalRoot || '.usa-modal-wrapper'
+
+  const NON_MODALS = `body > *:not(${modalRootSelector}):not([aria-hidden])`
+  const NON_MODALS_HIDDEN = `[data-modal-hidden]`
+
+  useEffect(() => {
+    const SCROLLBAR_WIDTH = getScrollbarWidth()
+    const INITIAL_PADDING =
+      window
+        .getComputedStyle(document.body)
+        .getPropertyValue('padding-right') || '0px'
+
+    const TEMPORARY_PADDING = `${
+      parseInt(INITIAL_PADDING.replace(/px/, ''), 10) +
+      parseInt(SCROLLBAR_WIDTH.replace(/px/, ''), 10)
+    }px`
+
+    initialPaddingRef.current = INITIAL_PADDING
+    tempPaddingRef.current = TEMPORARY_PADDING
+
+    setMounted(true)
+  }, [])
+
+  useEffect(() => {
+    if (mounted) {
+      const { body } = document
+
+      const INITIAL_PADDING = initialPaddingRef.current
+      const TEMPORARY_PADDING = tempPaddingRef.current
+
+      body.style.paddingRight =
+        (body.style.paddingRight === TEMPORARY_PADDING
+          ? INITIAL_PADDING
+          : TEMPORARY_PADDING) || ''
+
+      if (isOpen === true) {
+        body.classList.add('usa-js-modal--active')
+
+        document.querySelectorAll(NON_MODALS).forEach((el) => {
+          el.setAttribute('aria-hidden', 'true')
+          el.setAttribute('data-modal-hidden', '')
+        })
+
+        if (forceAction) {
+          body.classList.add('usa-js-no-click')
+        }
+      } else if (isOpen === false) {
+        body.classList.remove('usa-js-modal--active')
+        body.classList.remove('usa-js-no-click')
+
+        document.querySelectorAll(NON_MODALS_HIDDEN).forEach((el) => {
+          el.removeAttribute('aria-hidden')
+          el.removeAttribute('data-modal-hidden')
+        })
+      }
+    }
+
+    return () => {
+      const { body } = document
+
+      body.classList.remove('usa-js-modal--active')
+      body.classList.remove('usa-js-no-click')
+
+      document.querySelectorAll(NON_MODALS_HIDDEN).forEach((el) => {
+        el.removeAttribute('aria-hidden')
+        el.removeAttribute('data-modal-hidden')
+      })
+    }
+  }, [isOpen])
+
+  const ariaLabelledBy = divProps['aria-labelledby']
+  const ariaDescribedBy = divProps['aria-describedby']
+
+  delete divProps['aria-labelledby']
+  delete divProps['aria-describedby']
```

Comment:
My understanding of `delete` leads me to believe that `ariaLabelledBy` and `ariaDescribedBy` would always be undefined here, and therefore also when passed in to the `ModalWrapper`. Is that not the case?

Reply:
When I run storybook examples, they are indeed passed along appropriately. So this works, and I just don't understand why. I can trace through that in my own time to understand it 👍

Reply:
this is because these 2 props get applied to the `ModalWrapper` component explicitly, with the rest of the props being passed into the `ModalWindow`

Reply:
Yea, I just assumed that `ariaLabelledBy` would become `undefined` when `delete divProps['aria-labelledby']` runs, assuming that `ariaLabelledBy` would hold a reference to `divProps['aria-labelledby']`, but that's just me not knowing exactly how JS behaves 😅